website/app/components/doc/scroll-to-top/index.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+export default class DocScrollToTopComponent extends Component {
+  @service fastboot;
+  @tracked isButtonVisible = false;
+
+  constructor() {
+    super(...arguments);
+    this.addScrollListener();
+  }
+
+  willDestroy() {
+    super.willDestroy(...arguments);
+    this.removeScrollListener();
```

Comment:
consider adding the check here, to keep the `removeScrollListener` more agnostic
```suggestion
    if (!this.fastboot.isFastBoot) {
      this.removeScrollListener();
    }
```